### PR TITLE
Adding impl/test crumbs

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding impl/test crumbs for some random requirements.

## Why is the change being made?

This is part of on-going requirement work.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [ ] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/blocks.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def __deepcopy__(self, memo):
 
         return b
 
-    def _createHomogenizedCopy(self, pinSpatialLocators=False):
+    def createHomogenizedCopy(self, pinSpatialLocators=False):
```
Comment (keckler) [inline armi/reactor/blocks.py:160]: This (private) method is referenced in a docstring of one of our internal plugins. Not a huge deal, just FYI.

File: armi/reactor/tests/test_components.py [hunk 2]
```diff
@@ -217,13 +217,21 @@ def test_solid_material(self):
...
+        self.assertIn("HT9", str(self.component.getProperties()))
+
         self.component.material = air.Air()
         self.assertFalse(self.component.containsSolidMaterial())
 
         self.component.material = alloy200.Alloy200()
         self.assertTrue(self.component.containsSolidMaterial())
 
+        self.assertIn("Alloy200", str(self.component.getProperties()))
+
```
Comment (keckler) [inline armi/reactor/tests/test_components.py:234]: I think we probably need to take this a little further and show that there are actually properties here. I don't immediately see how testing for the material name is proving this.

File: armi/reactor/blocks.py (hunk 1 above)
Comment (john-science) [reply armi/reactor/blocks.py:160]: Ah, okay, I'll have to fix that too.

Can you point me at it?  I don't see it.

Comment (john-science) [reply armi/reactor/blocks.py:160]: Handled.  Thanks!

File: armi/reactor/tests/test_components.py (hunk 2 above)
Comment (keckler) [reply armi/reactor/tests/test_components.py:234]:
```suggestion

        .. test:: Components have material properties.
            :id: T_ARMI_COMP_MAT
            :tests: R_ARMI_COMP_MAT
        """
        self.assertTrue(hasattr(self.component.material, "density"))

        self.component.material = air.Air()
        self.assertFalse(self.component.containsSolidMaterial())

        self.component.material = alloy200.Alloy200()
        self.assertTrue(self.component.containsSolidMaterial())

        self.assertTrue(hasattr(self.component.material, "density"))

```

What do you think of that instead? I deleted the old asserts, but we could leave those in as well.

Comment (john-science) [reply armi/reactor/tests/test_components.py]: Okay, I added those. And kept the old ones.

**Pull request merged**